tests/type-check/bad/data-as-object-extend-missing.arr

```diff
@@ -0,0 +1,5 @@
+data Foo:
+  | bar(a :: Number, ref b :: String)
+end
+
+bar(1, "a").{a: "a"}
```

Comment:
Are these test cases named correctly?  This doesn't seem to be a missing field, it's a wrong type.  In general, could you add a comment to each bad test, above the failing expression, explaining what the error should be?

Reply:
Oops I flipped two of the test files for both update and extend.

Reply:
This looks almost right...but you forgot to fix this specific file.  After that, I'm good to merge this.

Reply:
I have fixed it now.